### ISSUE TITLE
ANDI hides the dropdown and pop up window on Linearizing the page

### ISSUE THREAD

**Issue by richa1920** [SSAgov/ANDI#195]
Hi,

It looks like the ANDI "Linearize Page" tool does not work properly with dropdowns and pop up windows on a page. Below is a situation where a pop window is hidden when we select the Linearize option from the ANDI Tool.
Here is a screenshot of before ANDI-linearized page
![image](https://user-images.githubusercontent.com/131186403/232840964-50c7e6e8-5323-4607-bd9f-6a7fd6509465.png)

Here is a screenshot of the ANDI-linearized page:
![image](https://user-images.githubusercontent.com/131186403/232841426-c23592cb-fbaa-485d-ac68-3351781b13f2.png)

However, when we try checking for the pop up elements by developer tool, we are able to find it(highlighted below)
![image](https://user-images.githubusercontent.com/131186403/232842865-709076b1-f4b8-4d79-9cb1-f4526a32f0f2.png)

So, basically it creates a whitespace and hides the element of the popover in the page.
Can anyone suggest what is going wrong here?

Thank you!

**Comment by JohnCotterSSA** [comment]
Hi, Here's some info from ANDI on what the linearize function does:
[Linearize](https://www.ssa.gov/accessibility/andi/help/howtouse.html#LinearizePage)

> This setting will "linearize" the page by searching for elements that have been positioned with CSS, and repositioning them in the order in which a screen reader would detect them.
> 
> Elements that have been repositioned will have a teal colored halo around them.
> 
> This can be useful when testing the meaningful sequence of elements on the page.

So if the dropdowns and popup windows are being positioned with CSS, then ANDI's linearize is changing that CSS property value, and it could definitely cause a display issue.

The linearize function is a troubleshooting function that intentionally causes displays to look different - and it's not always going to be pretty. You might be able to get what you're after if you use [sANDI Reading Order function](https://www.ssa.gov/accessibility/andi/help/modules.html#sANDI-readingOrder).

Let me know if that helps.

**Comment by richa1920** [comment]
Hi Thank you for the reply.
We understand that ANDI "linearize" the page by searching for elements that have been positioned with CSS, and repositioning them in the order in which a screen reader would detect them.
For our case we do see that on linearizing the page, popover is getting completely hidden but we do see the reading order as you mentioned previously for this as well.
![image](https://user-images.githubusercontent.com/131186403/234764754-fe1cf606-e9c0-477c-9546-69f850db3365.png)
 Also, I would like to mention two more points-
1. There are other tools that shows the pop up in the reading order on the page after Linearizing.
2. For the dropdowns, though the positions does get changed it does show it in the page in reading order upon Linearizing.

Please let me know if you need more information on this.

**Comment by JohnCotterSSA** [comment]
The screenshot you provided has ANDI tab order flags (yellow) not the reading order flags (purple). 

To help diagnose this problem further, we would need a working example of the issue.

**Comment by richa1920** [comment]
Hi,
As you mentioned earlier, I tried the reading order pane and it does show purple color reading order for the popover item as well as shown in below screenshot-
![image](https://user-images.githubusercontent.com/131186403/236792674-c205cfa4-b8c1-4d60-9620-57d5eeae2aca.png)
I need the below queries to be addressed by you-
1.why does it completely hide the popover and show whitespace in that place after repositioning it as shown below
![image](https://user-images.githubusercontent.com/131186403/236796361-9793932d-9925-447a-acf9-674172e428fc.png)

2.In dropdown case, it does reposition the dropdown in the reading order as shown below.
![image](https://user-images.githubusercontent.com/131186403/236796643-f1a6eab4-2e45-459d-80e3-f4929a99cc63.png)

Apologies, I cannot provide you the working example as of now but it would be great if you can let me know the answer of above queries.
However, I will try to provide the working example as soon as I can.

Thank you.
